capstone/capdb/models.py

```diff
@@ -540,13 +717,27 @@ def __str__(self):
     def get_slug(self):
         return self.cite[:100]
 
+
 class PageXML(BaseXMLModel):
     barcode = models.CharField(max_length=255, unique=True, db_index=True)
     volume = models.ForeignKey(VolumeXML, related_name='page_xmls',
-                                     on_delete=models.DO_NOTHING)
+                               on_delete=models.DO_NOTHING)
     cases = models.ManyToManyField(CaseXML, related_name='pages')
     s3_key = models.CharField(max_length=1024, blank=True, help_text="s3 path")
 
+    tracker = FieldTracker()
+
+    def save(self, force_insert=False, force_update=False, *args, **kwargs):
```

Comment:
This gets a bit recursive -- we'll have a Case save triggering a Page save triggering a Case save. Can we do this so each thing is only saved once? 

Maybe by adding options for it -- `update_case_md5 = kwargs.pop('update_case_md5'); update_volume_md5 = kwargs.pop('update_volume_md5');`. Then in Case.save() you can pass in False, and take care of it there. This would probably also mean adding a `save=True` option to `update_related_md5()` so you can update in Case.save without immediately saving.